src/main/java/org/opensearch/flowframework/transport/ReprovisionWorkflowTransportAction.java

```diff
@@ -270,27 +270,45 @@ private void executeWorkflowAsync(
         ActionListener<WorkflowResponse> listener
     ) {
         try {
-            threadPool.executor(PROVISION_WORKFLOW_THREAD_POOL).execute(() -> { executeWorkflow(template, workflowSequence, workflowId); });
+            threadPool.executor(PROVISION_WORKFLOW_THREAD_POOL).execute(() -> {
+                updateTemplate(template, workflowId);
+                executeWorkflow(workflowSequence, workflowId);
+            });
         } catch (Exception exception) {
             listener.onFailure(new FlowFrameworkException("Failed to execute workflow " + workflowId, ExceptionsHelper.status(exception)));
         }
     }
 
     /**
-     * Executes the given workflow sequence
+     * Replace template document
      * @param template The template to store after reprovisioning completes successfully
+     * @param workflowId The workflowId associated with the workflow that is executing
+     */
+    private void updateTemplate(Template template, String workflowId) {
+        flowFrameworkIndicesHandler.updateTemplateInGlobalContext(workflowId, template, ActionListener.wrap(templateResponse -> {
+            logger.info("Updated template for {}", workflowId, State.COMPLETED);
```

Comment:
The `State.COMPLETED` here is ignored as there's only one `{}` parameter. Which is a good thing because I thought it was out of place anyway, and should just be deleted.